### PR TITLE
feat: use controllable clock for processing metrics

### LINKED ISSUE #21051: Stream platform time is controllable
To support a [clock control API](https://github.com/camunda/product-hub/issues/2379), we will extend the stream platform.
The idea is that the engine uses and controls a partition-scoped clock that is then used to assign record timestamps, trigger scheduled tasks, and generally drive all business logic.

### Acceptance Criteria
- [x] All records are written with a controllable timestamp
- [x] Engine only uses controllable clock, no other sources of time
- [x] Scheduled tasks are executed according to the controllable clock
- [x] Controllable clock is only used within one partition

```[tasklist]
### Implementation
- [x] `RecordProcessorContext` contains a controllable clock
- [x] `Engine` uses controllable clock instead of `ActorClock`
- [x] `Engine` can modify the controllable clock by pinning or applying an offset
- [x] `Sequencer` uses controllable clock to assign record timestamps
- [x] `ProcessingScheduleService` uses controllable clock
- [x] `MetricsExporter` uses controllable clock
- [ ] `ProcessingStateMachine` uses controllable clock for latency metrics
```

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

This ensures that metrics published by the `ProcessingStateMachine` and the `MetricsExporter` are respecting the controllable clock.

To support this in the `MetricsExporter`, the `ExporterContext` was extended to provide an `InstantSource` that all exporters can use.

## Related issues

closes #21051

### CODE REVIEW

**Comment by npepinpe** [comment]
Does this mean that general exporters will now be able to read the engine clock? :thinking: It might be confusing since our underlying scheduler there will not be using the clock at all, but... :thinking:

File: zeebe/exporter-api/src/main/java/io/camunda/zeebe/exporter/api/context/Context.java [hunk 1]
```diff
@@ -30,6 +31,12 @@ public interface Context {
    */
   Logger getLogger();
 
+  /**
+   * @return a clock that provides the current time. Use this instead of system time to ensure that
```
Comment (npepinpe) [inline zeebe/exporter-api/src/main/java/io/camunda/zeebe/exporter/api/context/Context.java:35]: :question: Do we want to make it explicit that this is not related to the exporter's `Controller` task scheduling capabilities? I think it might cause confusion.

**Review by npepinpe — APPROVED** [review]
Looks good, just one question :thinking:

File: zeebe/exporter-api/src/main/java/io/camunda/zeebe/exporter/api/context/Context.java (hunk 1 above)
Comment (lenaschoenburg) [reply zeebe/exporter-api/src/main/java/io/camunda/zeebe/exporter/api/context/Context.java:35]: I'll add something :+1:
